spring-integration-core/src/main/java/org/springframework/integration/handler/DelayHandler.java

```diff
@@ -370,11 +457,75 @@ private void releaseMessageAfterDelay(final Message<?> message, long delay) {
 	}
 
 	private void releaseMessage(Message<?> message) {
-		this.releaseHandler.handleMessage(message);
+		String identity = ObjectUtils.getIdentityHexString(message);
+		this.deliveries.putIfAbsent(identity, new AtomicInteger());
+		try {
+			this.releaseHandler.handleMessage(message);
+			this.deliveries.remove(identity);
+		}
+		catch (Exception e) {
+			if (getErrorChannel() != null) {
+				ErrorMessage errorMessage = new ErrorMessage(e,
+						Collections.singletonMap(IntegrationMessageHeaderAccessor.DELIVERY_ATTEMPT,
+								new AtomicInteger(this.deliveries.get(identity).get() + 1)),
```

Comment:
I believe we just need to initialize an `AtomicInteger` with `1` and reuse it here. 
And, of course, we would need to use a `getAndIncrement()` below instead.
Or do you worry about mutation from the target error process?

What was the point though to make such a header mutable from the first place?
I don't see any mutations throughout the project 🤷‍♂️ 

WDYT?

Reply:
I don't want to give the error flow the ability to change the hard reference to the actual integer in the map. 

It's an `AtomicInteger` because the header accessor expects it to be.